src/components/forms/TextInput/TextInputMask.tsx

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable security/detect-object-injection */
+import React, { useState } from 'react'
+import classnames from 'classnames'
+import { TextInput, TextInputProps } from './TextInput'
+
+type MaskProps = {
+  mask: string
+  charset?: string
+}
+
+export type AllProps = TextInputProps & MaskProps
+
+export const TextInputMask = ({
+  id,
+  name,
+  type,
+  className,
+  validationStatus,
+  inputSize,
+  inputRef,
+  mask,
+  charset,
+  ...inputProps
+}: AllProps): React.ReactElement => {
+  const classes = classnames(
+    {
+      'usa-masked': mask,
+    },
+    className
+  )
+
+  const [inputValue, setInputValue] = useState('')
+  const [maskValue, setMaskValue] = useState(mask)
+  const [iValue, setIValue] =
+    useState<
+      React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
+    >()
```

Comment:
How about we just make this a string:

```suggestion
  const [iValue, setIValue] = useState('')
```